File: kobciye/lib/features/dashboards/widgets/section_placeholder.dart
import 'package:flutter/material.dart';
import '../../../core/constants/app_text.dart';
import '../../../services/localization_service.dart';

/// Clean "coming in a later phase" empty-state panel for nav sections that
/// don't have real content yet.
class SectionPlaceholder extends StatelessWidget {
  final String titleKey;
  final IconData icon;
  final Gradient gradient;

  const SectionPlaceholder({super.key, required this.titleKey, required this.icon, required this.gradient});

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(
            width: 88,
            height: 88,
            decoration: BoxDecoration(gradient: gradient, borderRadius: BorderRadius.circular(28)),
            child: Icon(icon, color: Colors.white, size: 38),
          ),
          const SizedBox(height: 22),
          Text(context.t(titleKey), style: AppText.h1),
          const SizedBox(height: 8),
          Text(context.t('comingSoon'), style: AppText.bodyMuted),
        ],
      ),
    );
  }
}

File: kobciye-rn/src/widgets/SectionPlaceholder.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Ionicons } from '@expo/vector-icons';
import { Text as TextStyles } from '../constants/text';
import { useLocalization } from '../context/LocalizationContext';

// Clean "coming in a later phase" empty-state panel for nav sections that
// don't have real content yet.
export default function SectionPlaceholder({ titleKey, icon, gradient }) {
  const { t } = useLocalization();
  return (
    <View style={styles.center}>
      <LinearGradient colors={gradient} style={styles.iconWrap}>
        <Ionicons name={icon} size={36} color="#fff" />
      </LinearGradient>
      <Text style={[TextStyles.h1, { marginTop: 22, marginBottom: 8 }]}>{t(titleKey)}</Text>
      <Text style={TextStyles.bodyMuted}>{t('comingSoon')}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  center: { flex: 1, alignItems: 'center', justifyContent: 'center', padding: 32 },
  iconWrap: { width: 88, height: 88, borderRadius: 28, alignItems: 'center', justifyContent: 'center' },
});
